internal/k8s/exec.go

```diff
@@ -170,14 +171,29 @@ func ReturnDeploymentObject(clientset *kubernetes.Clientset, matchLabel string,
 
 	log.Info().Msgf("waiting for %s Deployment to be created", matchLabelValue)
 
-	// Create watch operation
-	objWatch, err := clientset.
-		AppsV1().
-		Deployments(namespace).
-		Watch(context.Background(), deploymentListOptions)
-	if err != nil {
-		log.Error().Msgf("error when attempting to search for Deployment: %s", err)
-		return nil, err
+	var objWatch watch.Interface
+	var err error
+
+	timeout := 20 * time.Second
+	startTime := time.Now()
+
+	// Create watch operation with retries
+	for {
+		// Attempt to watch the deployment
+		objWatch, err = clientset.AppsV1().Deployments(namespace).Watch(context.Background(), deploymentListOptions)
+
+		if err == nil {
+			break
+		}
+
+		// Check if the timeout has been reached
+		if time.Since(startTime) > timeout {
+			log.Printf("Error when attempting to watch Deployment: %s", err)
+			return nil, fmt.Errorf("timeout reached while watching for Deployment")
+		}
+
+		log.Printf("Error when attempting to watch Deployment: %s. Retrying...", err)
+		time.Sleep(1 * time.Second)
```

Comment:
Oh man I've been debating whether or not I should rewrite all these 🤣 

So the standard Kubernetes `client-go` has a `wait` package that can be used instead of us writing these manually. I was trying to get a crack at it before but I was sinking too much time into it.